gapii/cc/spy.cpp

```diff
@@ -97,12 +108,34 @@ Spy::Spy()
       mObserveFrameFrequency(0),
       mObserveDrawFrequency(0),
       mFrameNumber(0) {
+  // Start by checking whether to capture the current process: compare the
+  // current process name with the "capture_proc_name" that we get from the
+  // environment. An empty "capture_proc_name" means capture any process. This
+  // is useful for games where the process initially started by AGI creates an
+  // other process where the actual game rendering happens.
   bool this_executable = true;
-  char* pn = getenv("GAPID_CAPTURE_PROCESS_NAME");
-  if (pn) {
-    auto proc_name = core::get_process_name();
-    this_executable = (proc_name == pn);
+  auto this_proc_name = core::get_process_name();
+  GAPID_INFO("this process name: %s", this_proc_name.c_str());
+
+#if TARGET_OS == GAPID_OS_ANDROID
+  // PROP_VALUE_MAX is defined in <sys/system_properties.h>
+  char capture_proc_name[PROP_VALUE_MAX];
+  __system_property_get(kCaptureProcessNameSystemProperty, capture_proc_name);
+#else
+  const char* capture_proc_name = getenv(kCaptureProcessNameEnvVar);
+  // Make sure capture_proc_name is not null, since the code below assumes so.
+  if (!capture_proc_name) {
+    capture_proc_name = "";
   }
+#endif
+
+  if (strlen(capture_proc_name)) {
```

Comment:
No need to count the length, just use `if (capture_proc_name != nullptr && *capture_proc_name != nullptr) {` to check if the string is valid and non-empty. Then you can also remove the 4 lines after the getenv command.

Reply:
I couldn't do `capture_proc_name != nullptr` with the Android code path since that led to:
```
gapii/cc/spy.cpp:132:8: error: comparison of array 'capture_proc_name' not equal to a null pointer is always true [-Werror,-Wtautological-pointer-compare]
```

But I agree that the code was not ideal, it's much easier to do a type cast to work around the compilation warning. I've updated accordingly, with a proper comment.